lib/core/commands/user-commands.js

```diff
@@ -7,31 +7,51 @@ class UserCommands {
   async createUser(user) {
     const command = {
       createUser: user.user,
-      // mongo internally uses pwd
-      pwd: user.password,
       customData: user.customData || {},
       roles: user.roles || []
     }
+
+    // mongo internally uses pwd
+    if (user.password) {
+      command.pwd = user.password
+    }
+    // Use external db iff external is set
+    if (user.external) {
+      this.client.useDatabase('$external')
+    }
+
     return await this.client.command(command)
   }
 
   // https://docs.mongodb.com/manual/reference/command/updateUser/
   async updateUser(user) {
     const command = {
       updateUser: user.user,
-      // mongo internally uses pwd
-      pwd: user.password,
       customData: user.customData || {},
       roles: user.roles || []
     }
+
+    // mongo internally uses pwd
+    if (user.password) {
```

Comment:
This is actually mutually exclusive with external

Reply:
I am aware but for now, I felt it was easier to leave this up to the user to put the right data in config just like the mongo API call supports both. 

I feel like once the JOI validation is in we validate this on a configuration layer too so it's more user friendly?